### ISSUE TITLE
TS server web file reads cause extension host starvation

### ISSUE THREAD

**Issue by joyceerhl** [microsoft/vscode#175972]
1. Open a GitHub PR in insiders.vscode.dev with `?vscode-coi=`
2. Wait for `Initializing JS/TS features` status bar item to go away
3. Make a change to the PR and commit it 
4. :bug: commit stalls for something like 30s with SCM viewlet progress. The whole time the GitHub Repositories output channel shows that file reads are happening. Sample of logs:

```
[2023-03-02 22:09:22:063] [  365] GitHub.getFileTree(github://github%2B7b2276223a312c22726566223a7b2274797065223a332c226964223a22313735373637227d7d/microsoft/vscode:eafb283e55f8b695b5f378a6b3c0e846c8c9fa09): github://github%2B7b2276223a312c22726566223a7b2274797065223a332c226964223a22313735373637227d7d/microsoft/vscode/src/vs/base/browser took 0 ms
[2023-03-02 22:09:22:074] [  366] GitHub.getFileTree(github://github%2B7b2276223a312c22726566223a7b2274797065223a332c226964223a22313735373637227d7d/microsoft/vscode:eafb283e55f8b695b5f378a6b3c0e846c8c9fa09): github://github%2B7b2276223a312c22726566223a7b2274797065223a332c226964223a22313735373637227d7d/microsoft/vscode/src/node_modules took 0 ms
[2023-03-02 22:09:22:074] [  367] GitHub.getFileTree(github://github%2B7b2276223a312c22726566223a7b2274797065223a332c226964223a22313735373637227d7d/microsoft/vscode:eafb283e55f8b695b5f378a6b3c0e846c8c9fa09): github://github%2B7b2276223a312c22726566223a7b2274797065223a332c226964223a22313735373637227d7d/microsoft/vscode/src/vs/base/browser took 0 ms
[2023-03-02 22:09:22:079] [  368] GitHub.getFileTree(github://github%2B7b2276223a312c22726566223a7b2274797065223a332c226964223a22313735373637227d7d/microsoft/vscode:eafb283e55f8b695b5f378a6b3c0e846c8c9fa09): github://github%2B7b2276223a312c22726566223a7b2274797065223a332c226964223a22313735373637227d7d/microsoft/vscode/node_modules took 0 ms
[2023-03-02 22:09:22:080] [  369] GitHub.getFileTree(github://github%2B7b2276223a312c22726566223a7b2274797065223a332c226964223a22313735373637227d7d/microsoft/vscode:eafb283e55f8b695b5f378a6b3c0e846c8c9fa09): github://github%2B7b2276223a312c22726566223a7b2274797065223a332c226964223a22313735373637227d7d/microsoft/vscode/src/vs/base/browser/dompurify took 0 ms
[2023-03-02 22:09:22:085] [  370] GitHub.getFileTree(github://github%2B7b2276223a312c22726566223a7b2274797065223a332c226964223a22313735373637227d7d/microsoft/vscode:eafb283e55f8b695b5f378a6b3c0e846c8c9fa09): github://github%2B7b2276223a312c22726566223a7b2274797065223a332c226964223a22313735373637227d7d/microsoft/vscode/src/vs/workbench/api/common/node_modules took 0 ms
[2023-03-02 22:09:22:086] [  371] GitHub.getFileTree(github://github%2B7b2276223a312c22726566223a7b2274797065223a332c226964223a22313735373637227d7d/microsoft/vscode:eafb283e55f8b695b5f378a6b3c0e846c8c9fa09): github://github%2B7b2276223a312c22726566223a7b2274797065223a332c226964223a22313735373637227d7d/microsoft/vscode/src/vs/base/browser/dompurify took 0 ms
[2023-03-02 22:09:22:088] [  372] GitHub.getFileTree(github://github%2B7b2276223a312c22726566223a7b2274797065223a332c226964223a22313735373637227d7d/microsoft/vscode:eafb283e55f8b695b5f378a6b3c0e846c8c9fa09): github://github%2B7b2276223a312c22726566223a7b2274797065223a332c226964223a22313735373637227d7d/microsoft/vscode/src/vs/workbench/api/node_modules took 0 ms
[2023-03-02 22:09:22:090] [  373] GitHub.getFileTree(github://github%2B7b2276223a312c22726566223a7b2274797065223a332c226964223a22313735373637227d7d/microsoft/vscode:eafb283e55f8b695b5f378a6b3c0e846c8c9fa09): github://github%2B7b2276223a312c22726566223a7b2274797065223a332c226964223a22313735373637227d7d/microsoft/vscode/src/vs/base/browser/ui took 0 ms
[2023-03-02 22:09:22:091] [  374] GitHub.getFileTree(github://github%2B7b2276223a312c22726566223a7b2274797065223a332c226964223a22313735373637227d7d/microsoft/vscode:eafb283e55f8b695b5f378a6b3c0e846c8c9fa09): github://github%2B7b2276223a312c22726566223a7b2274797065223a332c226964223a22313735373637227d7d/microsoft/vscode/src/vs/workbench/node_modules took 0 ms
[2023-03-02 22:09:22:114] [  375] GitHub.getFileTree(github://github%2B7b2276223a312c22726566223a7b2274797065223a332c226964223a22313735373637227d7d/microsoft/vscode:eafb283e55f8b695b5f378a6b3c0e846c8c9fa09): github://github%2B7b2276223a312c22726566223a7b2274797065223a332c226964223a22313735373637227d7d/microsoft/vscode/src/vs/node_modules took 0 ms
[2023-03-02 22:09:22:146] [  376] GitHub.getFileTree(github://github%2B7b2276223a312c22726566223a7b2274797065223a332c226964223a22313735373637227d7d/microsoft/vscode:eafb283e55f8b695b5f378a6b3c0e846c8c9fa09): github://github%2B7b2276223a312c22726566223a7b2274797065223a332c226964223a22313735373637227d7d/microsoft/vscode/src/vs/base/browser/ui took 0 ms
[2023-03-02 22:09:22:338] [  377] GitHub.getFileTree(github://github%2B7b2276223a312c22726566223a7b2274797065223a332c226964223a22313735373637227d7d/microsoft/vscode:eafb283e55f8b695b5f378a6b3c0e846c8c9fa09): github://github%2B7b2276223a312c22726566223a7b2274797065223a332c226964223a22313735373637227d7d/microsoft/vscode/src/node_modules took 0 ms
[2023-03-02 22:09:22:342] [  378] GitHub.getFileTree(github://github%2B7b2276223a312c22726566223a7b2274797065223a332c226964223a22313735373637227d7d/microsoft/vscode:eafb283e55f8b695b5f378a6b3c0e846c8c9fa09): github://github%2B7b2276223a312c22726566223a7b2274797065223a332c226964223a22313735373637227d7d/microsoft/vscode/src/vs/base/browser/ui/actionbar took 0 ms
[2023-03-02 22:09:22:361] [  379] GitHub.getFileTree(github://github%2B7b2276223a312c22726566223a7b2274797065223a332c226964223a22313735373637227d7d/microsoft/vscode:eafb283e55f8b695b5f378a6b3c0e846c8c9fa09): github://github%2B7b2276223a312c22726566223a7b2274797065223a332c226964223a22313735373637227d7d/microsoft/vscode/node_modules took 0 ms
[2023-03-02 22:09:22:370] [  380] GitHub.getFileTree(github://github%2B7b2276223a312c22726566223a7b2274797065223a332c226964223a22313735373637227d7d/microsoft/vscode:eafb283e55f8b695b5f378a6b3c0e846c8c9fa09): github://github%2B7b2276223a312c22726566223a7b2274797065223a332c226964223a22313735373637227d7d/microsoft/vscode/src/vs/base/browser/ui/actionbar took 0 ms
[2023-03-02 22:09:22:371] [  381] GitHub.getFileTree(github://github%2B7b2276223a312c22726566223a7b2274797065223a332c226964223a22313735373637227d7d/microsoft/vscode:eafb283e55f8b695b5f378a6b3c0e846c8c9fa09): github://github%2B7b2276223a312c22726566223a7b2274797065223a332c226964223a22313735373637227d7d/microsoft/vscode/src/vs/workbench/api/common/node_modules took 0 ms
[2023-03-02 22:09:22:375] [  382] GitHub.getFileTree(github://github%2B7b2276223a312c22726566223a7b2274797065223a332c226964223a22313735373637227d7d/microsoft/vscode:eafb283e55f8b695b5f378a6b3c0e846c8c9fa09): github://github%2B7b2276223a312c22726566223a7b2274797065223a332c226964223a22313735373637227d7d/microsoft/vscode/src/vs/workbench/api/node_modules took 0 ms
[2023-03-02 22:09:22:377] [  383] GitHub.getFileTree(github://github%2B7b2276223a312c22726566223a7b2274797065223a332c226964223a22313735373637227d7d/microsoft/vscode:eafb283e55f8b695b5f378a6b3c0e846c8c9fa09): github://github%2B7b2276223a312c22726566223a7b2274797065223a332c226964223a22313735373637227d7d/microsoft/vscode/src/vs/base/browser/ui/aria took 0 ms
[2023-03-02 22:09:22:379] [  384] GitHub.getFileTree(github://github%2B7b2276223a312c22726566223a7b2274797065223a332c226964223a22313735373637227d7d/microsoft/vscode:eafb283e55f8b695b5f378a6b3c0e846c8c9fa09): github://github%2B7b2276223a312c22726566223a7b2274797065223a332c226964223a22313735373637227d7d/microsoft/vscode/src/vs/workbench/node_modules took 0 ms
[2023-03-02 22:09:22:380] [  385] GitHub.getFileTree(github://github%2B7b2276223a312c22726566223a7b2274797065223a332c226964223a22313735373637227d7d/microsoft/vscode:eafb283e55f8b695b5f378a6b3c0e846c8c9fa09): github://github%2B7b2276223a312c22726566223a7b2274797065223a332c226964223a22313735373637227d7d/microsoft/vscode/src/vs/base/browser/ui/aria took 0 ms
[2023-03-02 22:09:22:386] [  386] GitHub.getFileTree(github://github%2B7b2276223a312c22726566223a7b2274797065223a332c226964223a22313735373637227d7d/microsoft/vscode:eafb283e55f8b695b5f378a6b3c0e846c8c9fa09): github://github%2B7b2276223a312c22726566223a7b2274797065223a332c226964223a22313735373637227d7d/microsoft/vscode/src/vs/node_modules took 0 ms
[2023-03-02 22:09:22:387] [  387] GitHub.getFileTree(github://github%2B7b2276223a312c22726566223a7b2274797065223a332c226964223a22313735373637227d7d/microsoft/vscode:eafb283e55f8b695b5f378a6b3c0e846c8c9fa09): github://github%2B7b2276223a312c22726566223a7b2274797065223a332c226964223a22313735373637227d7d/microsoft/vscode/src/vs/base/browser/ui/breadcrumbs took 0 ms
[2023-03-02 22:09:22:395] [  388] GitHub.getFileTree(github://github%2B7b2276223a312c22726566223a7b2274797065223a332c226964223a22313735373637227d7d/microsoft/vscode:eafb283e55f8b695b5f378a6b3c0e846c8c9fa09): github://github%2B7b2276223a312c22726566223a7b2274797065223a332c226964223a22313735373637227d7d/microsoft/vscode/src/node_modules took 0 ms
[2023-03-02 22:09:22:396] [  389] GitHub.getFileTree(github://github%2B7b2276223a312c22726566223a7b2274797065223a332c226964223a22313735373637227d7d/microsoft/vscode:eafb283e55f8b695b5f378a6b3c0e846c8c9fa09): github://github%2B7b2276223a312c22726566223a7b2274797065223a332c226964223a22313735373637227d7d/microsoft/vscode/src/vs/base/browser/ui/breadcrumbs took 0 ms
[2023-03-02 22:09:22:401] [  390] GitHub.getFileTree(github://github%2B7b2276223a312c22726566223a7b2274797065223a332c226964223a22313735373637227d7d/microsoft/vscode:eafb283e55f8b695b5f378a6b3c0e846c8c9fa09): github://github%2B7b2276223a312c22726566223a7b2274797065223a332c226964223a22313735373637227d7d/microsoft/vscode/node_modules took 0 ms
[2023-03-02 22:09:22:402] [  391] GitHub.getFileTree(github://github%2B7b2276223a312c22726566223a7b2274797065223a332c226964223a22313735373637227d7d/microsoft/vscode:eafb283e55f8b695b5f378a6b3c0e846c8c9fa09): github://github%2B7b2276223a312c22726566223a7b2274797065223a332c226964223a22313735373637227d7d/microsoft/vscode/src/vs/base/browser/ui/button took 0 ms
[2023-03-02 22:09:22:410] [  392] GitHub.getFileTree(github://github%2B7b2276223a312c22726566223a7b2274797065223a332c226964223a22313735373637227d7d/microsoft/vscode:eafb283e55f8b695b5f378a6b3c0e846c8c9fa09): github://github%2B7b2276223a312c22726566223a7b2274797065223a332c226964223a22313735373637227d7d/microsoft/vscode/src/vs/workbench/api/common/node_modules took 0 ms
[2023-03-02 22:09:22:411] [  393] GitHub.getFileTree(github://github%2B7b2276223a312c22726566223a7b2274797065223a332c226964223a22313735373637227d7d/microsoft/vscode:eafb283e55f8b695b5f378a6b3c0e846c8c9fa09): github://github%2B7b2276223a312c22726566223a7b2274797065223a332c226964223a22313735373637227d7d/microsoft/vscode/src/vs/base/browser/ui/button took 0 ms
[2023-03-02 22:09:22:415] [  394] GitHub.getFileTree(github://github%2B7b2276223a312c22726566223a7b2274797065223a332c226964223a22313735373637227d7d/microsoft/vscode:eafb283e55f8b695b5f378a6b3c0e846c8c9fa09): github://github%2B7b2276223a312c22726566223a7b2274797065223a332c226964223a22313735373637227d7d/microsoft/vscode/src/vs/workbench/api/node_modules took 0 ms
[2023-03-02 22:09:22:437] [  395] GitHub.getFileTree(github://github%2B7b2276223a312c22726566223a7b2274797065223a332c226964223a22313735373637227d7d/microsoft/vscode:eafb283e55f8b695b5f378a6b3c0e846c8c9fa09): github://github%2B7b2276223a312c22726566223a7b2274797065223a332c226964223a22313735373637227d7d/microsoft/vscode/src/vs/base/browser/ui/centered took 0 ms
[2023-03-02 22:09:22:440] [  396] GitHub.getFileTree(github://github%2B7b2276223a312c22726566223a7b2274797065223a332c226964223a22313735373637227d7d/microsoft/vscode:eafb283e55f8b695b5f378a6b3c0e846c8c9fa09): github://github%2B7b2276223a312c22726566223a7b2274797065223a332c226964223a22313735373637227d7d/microsoft/vscode/src/vs/workbench/node_modules took 0 ms
[2023-03-02 22:09:22:441] [  397] GitHub.getFileTree(github://github%2B7b2276223a312c22726566223a7b2274797065223a332c226964223a22313735373637227d7d/microsoft/vscode:eafb283e55f8b695b5f378a6b3c0e846c8c9fa09): github://github%2B7b2276223a312c22726566223a7b2274797065223a332c226964223a22313735373637227d7d/microsoft/vscode/src/vs/base/browser/ui/centered took 0 ms
[2023-03-02 22:09:22:442] [  398] GitHub.getFileTree(github://github%2B7b2276223a312c22726566223a7b2274797065223a332c226964223a22313735373637227d7d/microsoft/vscode:eafb283e55f8b695b5f378a6b3c0e846c8c9fa09): github://github%2B7b2276223a312c22726566223a7b2274797065223a332c226964223a22313735373637227d7d/microsoft/vscode/src/vs/node_modules took 0 ms
[2023-03-02 22:09:22:443] [  399] GitHub.getFileTree(github://github%2B7b2276223a312c22726566223a7b2274797065223a332c226964223a22313735373637227d7d/microsoft/vscode:eafb283e55f8b695b5f378a6b3c0e846c8c9fa09): github://github%2B7b2276223a312c22726566223a7b2274797065223a332c226964223a22313735373637227d7d/microsoft/vscode/src/vs/base/browser/ui/codicons took 0 ms
[2023-03-02 22:09:22:444] [  400] GitHub.getFileTree(github://github%2B7b2276223a312c22726566223a7b2274797065223a332c226964223a22313735373637227d7d/microsoft/vscode:eafb283e55f8b695b5f378a6b3c0e846c8c9fa09): github://github%2B7b2276223a312c22726566223a7b2274797065223a332c226964223a22313735373637227d7d/microsoft/vscode/src/node_modules took 0 ms

```

**Comment by joyceerhl** [comment]
This only happens with `vscode-coi` which is why I suspect an issue with the TS server support for web. For some reason there seems to be a lot of repeated reads to the same resources so I'm wondering if they can be avoided. If the reads really do need to happen, perhaps the server can yield every once in a while to allow other ext host tasks to proceed?

**Comment by joyceerhl** [comment]
Now that COI is enabled on insiders.vscode.dev this issue occurs constantly :/

**Comment by mjbvz** [comment]
I can't reproduce the stall. #183482 should fix the loading indicator showing forever though, however that was just a UI issue 

@joyceerhl Can you test again on the latest insiders and let me know if you are still seeing issues? If so, can you please share an example PR and steps to reproduce this? I've been testing using https://insiders.vscode.dev/github/microsoft/vscode/pull/183488

**Comment by joyceerhl** [comment]
I can still repro this on https://insiders.vscode.dev/github/microsoft/vscode/tree/dev/joyceerhl/still-mouse (just made a small commit and with the profiler running it took over 60 seconds in wall time)

FWIW I see this in the TS logs after enabling trace logging 
```
2023-05-26 13:56:45.732 [info] Starting TS Server undefined
2023-05-26 13:56:45.732 [info] Using tsserver from: https://main.vscode-cdn.net/insider/2dfb838f494f035099e999f0cd0eff5f1f488a30/extensions/typescript-language-features/dist/browser/typescript/tsserver.web.js undefined
2023-05-26 13:56:45.732 [info] <syntax> Forking...
2023-05-26 13:56:45.732 [info] <syntax> Starting...
2023-05-26 13:56:45.732 [info] <semantic> Forking...
2023-05-26 13:56:45.732 [info] <semantic> Starting...
2023-05-26 13:57:53.574 [trace] <semantic> Canceled request with sequence number 24
2023-05-26 13:57:53.577 [trace] <semantic> Canceled request with sequence number 25
   "file": "/review/github+7b2276223a312c22726566223a7b2274797065223a342c226964223a226465762f6a6f7963656572686c2f76697369626c652d77696c64656265657374227d7d/microsoft/vscode/src/vscode-dts/vscode.proposed.shareProvider.d.ts",
            "start": {
                "line": 63,
                "offset": 69
            },
            "end": {
                "line": 63,
                "offset": 77
            },
            "contextStart": {
                "line": 63,
                "offset": 69
            },
            "contextEnd": {
                "line": 63,
                "offset": 92
            }
        }
    ],
    "textSpan": {
        "start": {
            "line": 63,
            "offset": 69
        },
        "end": {
            "line": 63,
            "offset": 77
        }
    }
}

```

And then I see a huge number of error logs like this in the console

```
6  no watcher found for /github/github+7b2276223a312c22726566223a7b2274797065223a342c226964223a226465762f6a6f7963656572686c2f7374696c6c2d6d6f757365227d7d/microsoft/vscode/src/vs/workbench/services/workspaces/electron-sandbox
(anonymous) @ tsserver.web.js:6
D.f.onmessage @ tsserver.web.js:6
tsserver.web.js:6  no watcher found for /github/github+7b2276223a312c22726566223a7b2274797065223a342c226964223a226465762f6a6f7963656572686c2f7374696c6c2d6d6f757365227d7d/microsoft/vscode/src/vs/workbench/services/workspaces/test/browser
(anonymous) @ tsserver.web.js:6
D.f.onmessage @ tsserver.web.js:6
tsserver.web.js:6  no watcher found for /github/github+7b2276223a312c22726566223a7b2274797065223a342c226964223a226465762f6a6f7963656572686c2f7374696c6c2d6d6f757365227d7d/microsoft/vscode/src/vs/workbench/services/workspaces/test/common
(anonymous) @ tsserver.web.js:6
D.f.onmessage @ tsserver.web.js:6
tsserver.web.js:6  no watcher found for /github/github+7b2276223a312c22726566223a7b2274797065223a342c226964223a226465762f6a6f7963656572686c2f7374696c6c2d6d6f757365227d7d/microsoft/vscode/src/vs/workbench/test/browser/parts/editor
(anonymous) @ tsserver.web.js:6
D.f.onmessage @ tsserver.web.js:6
tsserver.web.js:6  no watcher found for /github/github+7b2276223a312c22726566223a7b2274797065223a342c226964223a226465762f6a6f7963656572686c2f7374696c6c2d6d6f757365227d7d/microsoft/vscode/src/vs/workbench/test/browser/parts/statusbar
(anonymous) @ tsserver.web.js:6
D.f.onmessage @ tsserver.web.js:6
tsserver.web.js:6  no watcher found for /github/github+7b2276223a312c22726566223a7b2274797065223a342c226964223a226465762f6a6f7963656572686c2f7374696c6c2d6d6f757365227d7d/microsoft/vscode/src/vs/workbench/test/common
(anonymous) @ tsserver.web.js:6
D.f.onmessage @ tsserver.web.js:6
tsserver.web.js:6  no watcher found for /github/github+7b2276223a312c22726566223a7b2274797065223a342c226964223a226465762f6a6f7963656572686c2f7374696c6c2d6d6f757365227d7d/microsoft/vscode/src/vs/workbench/test/electron-sandbox
(anonymous) @ tsserver.web.js:6
D.f.onmessage @ tsserver.web.js:6
tsserver.web.js:6  no watcher found for /github/github+7b2276223a312c22726566223a7b2274797065223a342c226964223a226465762f6a6f7963656572686c2f7374696c6c2d6d6f757365227d7d/microsoft/vscode/src/tsconfig.json
(anonymous) @ tsserver.web.js:6
D.f.onmessage @ tsserver.web.js:6
tsserver.web.js:6  no watcher found for /github/github+7b2276223a312c22726566223a7b2274797065223a342c226964223a226465762f6a6f7963656572686c2f7374696c6c2d6d6f757365227d7d/microsoft/vscode/src/tsconfig.base.json
(anonymous) @ tsserver.web.js:6
D.f.onmessage @ tsserver.web.js:6
tsserver.web.js:6  no watcher found for /github/github+7b2276223a312c22726566223a7b2274797065223a342c226964223a226465762f6a6f7963656572686c2f7374696c6c2d6d6f757365227d7d/microsoft/vscode/src/vscode-dts
(anonymous) @ tsserver.web.js:6
D.f.onmessage @ tsserver.web.js:6
tsserver.web.js:6  no watcher found for /github/github+7b2276223a312c22726566223a7b2274797065223a342c226964223a226465762f6a6f7963656572686c2f7374696c6c2d6d6f757365227d7d/microsoft/vscode/src/bootstrap.js
(anonymous) @ tsserver.web.js:6
D.f.onmessage @ tsserver.web.js:6
tsserver.web.js:6  no watcher found for /github/github+7b2276223a312c22726566223a7b2274797065223a342c226964223a226465762f6a6f7963656572686c2f7374696c6c2d6d6f757365227d7d/microsoft/vscode/src/bootstrap-amd.js
(anonymous) @ tsserver.web.js:6
D.f.onmessage @ tsserver.web.js:6
tsserver.web.js:6  no watcher found for /github/github+7b2276223a312c22726566223a7b2274797065223a342c226964223a226465762f6a6f7963656572686c2f7374696c6c2d6d6f757365227d7d/microsoft/vscode/src/server-main.js
(anonymous) @ tsserver.web.js:6
D.f.onmessage @ tsserver.web.js:6
tsserver.web.js:6  no watcher found for /github/github+7b2276223a312c22726566223a7b2274797065223a342c226964223a226465762f6a6f7963656572686c2f7374696c6c2d6d6f757365227d7d/microsoft/vscode/src
(anonymous) @ tsserver.web.js:6
D.f.onmessage @ tsserver.web.js:6
tsserver.web.js:6  no watcher found for /github/github+7b2276223a312c22726566223a7b2274797065223a342c226964223a226465762f6a6f7963656572686c2f7374696c6c2d6d6f757365227d7d/microsoft/vscode/src/vs
(anonymous) @ tsserver.web.js:6
D.f.onmessage @ tsserver.web.js:6
tsserver.web.js:6  no watcher found for /github/github+7b2276223a312c22726566223a7b2274797065223a342c226964223a226465762f6a6f7963656572686c2f7374696c6c2d6d6f757365227d7d/microsoft/vscode/src/bootstrap-node.js
(anonymous) @ tsserver.web.js:6
D.f.onmessage @ tsserver.web.js:6
tsserver.web.js:6  no watcher found for /github/github+7b2276223a312c22726566223a7b2274797065223a342c226964223a226465762f6a6f7963656572686c2f7374696c6c2d6d6f757365227d7d/microsoft/vscode/src/typings
(anonymous) @ tsserver.web.js:6
D.f.onmessage @ tsserver.web.js:6
tsserver.web.js:6  no watcher found for /github/github+7b2276223a312c22726566223a7b2274797065223a342c226964223a226465762f6a6f7963656572686c2f7374696c6c2d6d6f757365227d7d/microsoft/vscode/product.json
(anonymous) @ tsserver.web.js:6
D.f.onmessage @ tsserver.web.js:6
tsserver.web.js:6  no watcher found for /github/github+7b2276223a312c22726566223a7b2274797065223a342c226964223a226465762f6a6f7963656572686c2f7374696c6c2d6d6f757365227d7d/microsoft/vscode/package.json
(anonymous) @ tsserver.web.js:6
D.f.onmessage @ tsserver.web.js:6
tsserver.web.js:6  no watcher found for /github/github+7b2276223a312c22726566223a7b2274797065223a342c226964223a226465762f6a6f7963656572686c2f7374696c6c2d6d6f757365227d7d/microsoft/vscode/src/bootstrap-node.js
(anonymous) @ tsserver.web.js:6
D.f.onmessage @ tsserver.web.js:6
tsserver.web.js:6  no watcher found for /github/github+7b2276223a312c22726566223a7b2274797065223a342c226964223a226465762f6a6f7963656572686c2f7374696c6c2d6d6f757365227d7d/microsoft/vscode/src/typings/require.d.ts
(anonymous) @ tsserver.web.js:6
D.f.onmessage @ tsserver.web.js:6
tsserver.web.js:6  no watcher found for /github/github+7b2276223a312c22726566223a7b2274797065223a342c226964223a226465762f6a6f7963656572686c2f7374696c6c2d6d6f757365227d7d/microsoft/vscode/src/typings/thenable.d.ts
(anonymous) @ tsserver.web.js:6
D.f.onmessage @ tsserver.web.js:6
tsserver.web.js:6  no watcher found for /github/github+7b2276223a312c22726566223a7b2274797065223a342c226964223a226465762f6a6f7963656572686c2f7374696c6c2d6d6f757365227d7d/microsoft/vscode/src/typings/vscode-globals-modules.d.ts
(anonymous) @ tsserver.web.js:6
D.f.onmessage @ tsserver.web.js:6
tsserver.web.js:6  no watcher found for /github/github+7b2276223a312c22726566223a7b2274797065223a342c226964223a226465762f6a6f7963656572686c2f7374696c6c2d6d6f757365227d7d/microsoft/vscode/src/typings/vscode-globals-product.d.ts
(anonymous) @ tsserver.web.js:6
D.f.onmessage @ tsserver.web.js:6
tsserver.web.js:6  no watcher found for /github/github+7b2276223a312c22726566223a7b2274797065223a342c226964223a226465762f6a6f7963656572686c2f7374696c6c2d6d6f757365227d7d/microsoft/vscode/src/typings/windows-mutex.d.ts
(anonymous) @ tsserver.web.js:6
D.f.onmessage @ tsserver.web.js:6
tsserver.web.js:6  no watcher found for /github/github+7b2276223a312c22726566223a7b2274797065223a342c226964223a226465762f6a6f7963656572686c2f7374696c6c2d6d6f757365227d7d/microsoft/vscode/src/typings/windows-process-tree.d.ts
(anonymous) @ tsserver.web.js:6
D.f.onmessage @ tsserver.web.js:6
tsserver.web.js:6  no watcher found for /github/github+7b2276223a312c22726566223a7b2274797065223a342c226964223a226465762f6a6f7963656572686c2f7374696c6c2d6d6f757365227d7d/microsoft/vscode/src/typings/windows-registry.d.ts
(anonymous) @ tsserver.web.js:6
D.f.onmessage @ tsserver.web.js:6
tsserver.web.js:6  no watcher found for /github/github+7b2276223a312c22726566223a7b2274797065223a342c226964223a226465762f6a6f7963656572686c2f7374696c6c2d6d6f757365227d7d/microsoft/vscode/src/typings
(anonymous) @ tsserver.web.js:6
D.f.onmessage @ tsserver.web.js:6
tsserver.web.js:6  no watcher found for /github/github+7b2276223a312c22726566223a7b2274797065223a342c226964223a226465762f6a6f7963656572686c2f7374696c6c2d6d6f757365227d7d/microsoft/vscode/package.json
(anonymous) @ tsserver.web.js:6
```

**Comment by joyceerhl** [comment]
😨 these are a lot of errors, I could see the logging overhead potentially accounting for the perf degradation

<img width="99" alt="image" src="https://github.com/microsoft/vscode/assets/30305945/f990d711-b4e4-48ef-a686-f8fcfcced22a">

The URIs above also seem potentially malformed (`/github/github+XYZ`)

**Comment by mjbvz** [comment]
This seems to be fixed by #185920. The main downside is that we then have to redownload the repository index after the commit, but this is relatively quick

**Comment by joyceerhl** [comment]
Thank you! 🏆
